src/tests/functional/inference_engine/transformations/op_conversions/eye_decomposition_test.cpp

```diff
@@ -0,0 +1,286 @@
+        constructor_validate_and_infer_types();
+    }
+
+    std::shared_ptr<ov::Node> clone_with_new_inputs(const ov::OutputVector& new_args) const override {
+        check_new_args_count(this, new_args);
+        if (new_args.size() == 3) {
+            return std::make_shared<FakeEye>(new_args[0], new_args[1], new_args[2], ov::element::f32);
+        } else if (new_args.size() == 4) {
+            return std::make_shared<FakeEye>(new_args[0], new_args[1], new_args[2], new_args[3], ov::element::f32);
+        } else {
+            throw ov::Exception("FakeEye has incorrect input number: " + std::to_string(new_args.size()));
+        }
+    }
+};
+
+class EyeTransformationTests : public TransformationTestsF {
+protected:
+    EyeDecompositionWrapper eye_decomposition_wrapper;
+
+    ov::element::Type dtype;
+    size_t h, w;
+    int shift;
+
+    void SetUp() override {
+        TransformationTestsF::SetUp();
+
+        dtype = ov::element::f32;
+        h = 4;
+        w = 4;
+    }
+
+    template <class TEye>
+    std::shared_ptr<TEye> make_test_eye(const ov::Output<ov::Node>& k) const {
+        auto height = ov::op::v0::Constant::create(ov::element::i64, ov::Shape{1}, {h});
+        auto width = ov::op::v0::Constant::create(ov::element::i64, ov::Shape{1}, {w});
+
+        return std::make_shared<TEye>(height, width, k, dtype);
+    }
+
+    template <class TEye>
+    std::shared_ptr<TEye> make_test_eye() const {
+        auto k = ov::op::v0::Constant::create(ov::element::i64, ov::Shape{1}, {shift});
+
+        return make_test_eye<TEye>(k);
+    }
+
+    template <class TEye>
+    std::shared_ptr<TEye> make_test_eye_batch(const ov::Output<ov::Node>& batch) const {
+        auto height = ov::op::v0::Constant::create(ov::element::i64, ov::Shape{1}, {h});
+        auto width = ov::op::v0::Constant::create(ov::element::i64, ov::Shape{1}, {w});
+        auto k = ov::op::v0::Constant::create(ov::element::i64, ov::Shape{1}, {shift});
+
+        return std::make_shared<TEye>(height, width, k, batch, dtype);
+    }
+};
+
+/** \brief Diagonal shift is not `Constant`, there should be no decompose. */
+TEST_F(EyeTransformationTests, shift_is_not_const) {
+    {
+        auto data = std::make_shared<ov::op::v0::Parameter>(dtype, ov::Shape{h, w});
+        auto k = std::make_shared<ov::op::v0::Parameter>(ov::element::i64, ov::Shape{1});
+        auto node = make_test_eye<ov::op::v9::Eye>(k);
+
+        model = std::make_shared<ov::Model>(ov::NodeVector{node}, ov::ParameterVector{data, k});
+
+        manager.register_pass<ov::pass::EyeDecomposition>();
+    }
+
+    {
+        auto data = std::make_shared<ov::op::v0::Parameter>(dtype, ov::Shape{h, w});
+        auto k = std::make_shared<ov::op::v0::Parameter>(ov::element::i64, ov::Shape{1});
+        auto node = make_test_eye<ov::op::v9::Eye>(k);
+
+        model_ref = std::make_shared<ov::Model>(ov::NodeVector{node}, ov::ParameterVector{data, k});
+    }
```

Comment:
To tests `no decomposition` case, the reference part can be simply skipped.
```suggestion

```

Reply:
Removed.